x-pack/plugins/maps/public/routing/routes/create_edit/create_edit.js

```diff
@@ -0,0 +1,502 @@
+    const { refreshConfig } = this.state;
+    const newRefreshConfig = {
+      isPaused,
+      interval: refreshInterval ? refreshInterval : refreshConfig.interval,
+    };
+    this.setState({ refreshConfig: newRefreshConfig }, this.syncAppAndGlobalState);
+    this.props.setRefreshConfig(newRefreshConfig);
+  };
+
+  updateStateFromSavedQuery(savedQuery) {
+    const { filterManager } = getData().query;
+    const savedQueryFilters = savedQuery.attributes.filters || [];
+    const globalFilters = filterManager.getGlobalFilters();
+    const allFilters = [...savedQueryFilters, ...globalFilters];
+
+    if (savedQuery.attributes.timefilter) {
+      if (savedQuery.attributes.timefilter.refreshInterval) {
+        this.onRefreshChange({
+          isPaused: savedQuery.attributes.timefilter.refreshInterval.pause,
+          refreshInterval: savedQuery.attributes.timefilter.refreshInterval.value,
+        });
+      }
+      this.onQueryChange({
+        filters: allFilters,
+        query: savedQuery.attributes.query,
+        time: savedQuery.attributes.timefilter,
+      });
+    } else {
+      this.onQueryChange({
+        filters: allFilters,
+        query: savedQuery.attributes.query,
+      });
+    }
+  }
+
+  syncStoreAndGetFilters(savedMap) {
+    const { setGotoWithCenter, setMapSettings, setIsLayerTOCOpen, setOpenTOCDetails } = this.props;
+    let savedObjectFilters = [];
+    if (savedMap.mapStateJSON) {
+      const mapState = JSON.parse(savedMap.mapStateJSON);
+      setGotoWithCenter({
+        lat: mapState.center.lat,
+        lon: mapState.center.lon,
+        zoom: mapState.zoom,
+      });
+      if (mapState.filters) {
+        savedObjectFilters = mapState.filters;
+      }
+      if (mapState.settings) {
+        setMapSettings(mapState.settings);
+      }
+    }
+
+    if (savedMap.uiStateJSON) {
+      const uiState = JSON.parse(savedMap.uiStateJSON);
+      setIsLayerTOCOpen(_.get(uiState, 'isLayerTOCOpen', DEFAULT_IS_LAYER_TOC_OPEN));
+      setOpenTOCDetails(_.get(uiState, 'openTOCDetails', []));
+    }
+    return savedObjectFilters;
+  }
+
+  async initMap(savedMap) {
+    const { clearUi } = this.props;
+    // TODO: Handle null when converting to TS
+    const globalState = getGlobalState();
+    this.initMapAndLayerSettings(savedMap);
+    clearUi();
+
+    await this.handleStoreChanges();
+    this.storeSyncUnsubscribe = getStoreSyncSubscription(this.handleStoreChanges);
+
+    const savedObjectFilters = this.syncStoreAndGetFilters(savedMap);
+    await this.onQueryChange({
+      filters: [
+        ..._.get(globalState, 'filters', []),
+        ...this.getAppStateFilters(),
+        ...savedObjectFilters,
+      ],
+    });
+    this.setState({ initialized: true });
+  }
+
+  _renderTopNav() {
+    const {
+      query,
+      time,
+      refreshConfig,
+      savedMap,
+      initialLayerListConfig,
+      isVisible,
+      indexPatterns,
+      isSaveDisabled,
+      currentPath,
+    } = this.state;
+
```

Comment:
How about returning null here if `isVisible` is false. There is really no point in passing `isVisible` to `MapsTopNavMenu` just to have it return null later in the call stack. That way you will be passing one less thing to `MapsTopNavMenu` and simplify that component just a bit.